packages/react/src/W3iWidget/index.tsx

```diff
@@ -25,7 +31,24 @@ const htmlifyParams = (
 };
 
 const W3iWidget: React.FC<W3iWidgetProps> = (props) => {
-  return <w3i-widget {...htmlifyParams(props)} />;
+  const { signMessage } = props;
+  const spanRef = useRef<HTMLSpanElement>(null);
+
+  useEffect(() => {
+    if (!spanRef.current) return;
+
+    const w3iWidget: any = spanRef.current.firstChild;
```

Comment:
Would be great if you could add a contextual comment on why we take this approach via the ref, the way you explained to me in the huddle 👍